SparkPost/2web2ui
# SA-1957: Added deliverability metrics to the metrics drawer

### What Changed

- Add deliverability metrics
- Add deliverability data source to group by table and compare by group by table
- Added reusable component for menu dropdown
- Added reusable component for checkbox group with selectAll

### How To Test

- Go to report builder
- Verify that deliverability dropdown and banner do not exist on new account
- Verify that deliverability dropdown exist (with metrics disabled) and banner exist on account with feature flag `allow_deliverability_metrics`
- Verify that deliverability dropdown exist (with metrics enabled) and banner don't exist on account with feature flag and billing product
- Apply deliverability metrics
- Verify that group by table has addition dropdown when selecting a group by

### To Do

- [x] Add testing
- [x] Handle placement of top domains checkbox
- [x] Left align vs right align checkbox in mock
- [x] Link to enable data sources?
- [x] Follow up with possible weird state behavior (sending metrics, but d12y data sources)
